course/functions.php

```diff
@@ -101,14 +101,56 @@ function course_save_global_styles( $post_id, $post, $update ) {
 	$global_styles      = json_decode( $post->post_content, true );
 	$global['settings'] = isset( $global_styles['settings'] ) ? $global_styles['settings'] : array();
 	$global['styles']   = isset( $global_styles['styles'] ) ? $global_styles['styles'] : array();
+
+	if ( empty( $global['settings'] ) && empty( $global['styles'] ) ) {
+		update_option( 'course_theme_variation', 'default' );
+		return;
+	}
+
+	$td                 = WP_Theme_JSON_Resolver::get_theme_data();
 	$variations         = WP_Theme_JSON_Resolver::get_style_variations();
-	$current_variation  = 'default';
+	$variations[]       = array(
```

Comment:
Not really needed as we catch with empty values.
But added it in case behavior changes.